### ISSUE TITLE
Flow does not validate shorthand true properties in JSX

### ISSUE THREAD

**Issue by Li0liQ** [facebook/flow#4578]
If you pass `true` as a shorthand prop in JSX, flow will not perform type checks.
```
/* @flow */
import React from 'react';

type Props = {
  test: 'a'
};

class Test extends React.Component<void, Props, void> {
  render() {
  	return <div>{this.props.test}</div>
  }
}

const t = <Test test />; // This should fail, as test is expected to be 'a' and not true
```
[flow-try](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAlgWwA5wE4AuYASgKYCGAxkVHnJmAOR4XWMDcqqBAntqWAAKdbAGcwAXjABvVGDAFSoggC4m5RqgC+nVJRjlR4gCpKipAB6KAdgBNxZKgQB0AYXq5rpawQA8ANzh0WwAaIRFRMMDggD4ZOTAWO1I8AAoASnj5AEgWAgBXPGswX1t0fxjpAgALdFFnbAjnRWUtX2AyioStbS5KOGtlBUkS0yGWomAY9jBgYDBjWvFRarh8mFswKHJ0GDDDBTMwOrBLfmpSTYI4MAAjAUYNMHI7MGs4IgI8fNIgA)

**Comment by asolove** [comment]
I think this is fixed in v0.53.1. I had to make one minor change because the `React.Component` signature changed, [but that code](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAlgWwA5wE4AuYASgKYCGAxkVHnJmAOR4XWMDcqqBAntqWAAKdbAGcwAXjABvVGDAFSoggC4m5RqgC+nVJRjlR4gCpKipAB6KAdgBNxZKgQB0AYXq5rpawQA8wuDEAGjAANzh0WwA+GTkwFjtSPAAKAEpY+QBIFgIAVzxrMF9bdFCo6QIAC3RRZ2wRWsVlLV9gErK4rW0uSjhrZQVJItMBpqJgKPYwYGAwY2rxUUq4XJhbMChydBgQwwUzMBqwS35qUnWCODAAIwFGDTByOzBrOCICPFzSIA) now returns the type error you'd expect:

    14: const t = <Test test />; // This should fail, as test is expected to be 'a' and not true
                  ^ props of React element `Test`. This type is incompatible with
    8: class Test extends React.Component<Props, void> {
                                          ^ object type

I think this issue can be closed.

**Comment by Li0liQ** [comment]
Closing the issue as it has been fixed in flow v0.53.0.
